### ISSUE TITLE
Add customisable heading level to FAQ.Question component

### ISSUE THREAD

**Issue by rezrah** [primer/brand#191]
FAQ component headings use `h4` elements, which aren't currently customisable. 

The inability to change this becomes problematic when `h4` is not appropriate during integration.

AXE will also report a violation of heading order, which is one of the consequences of not being able to customise this value.

```json
{
    "id": "heading-order",
    "impact": "moderate",
    "tags": [
      "cat.semantics",
      "best-practice"
    ],
    "description": "Ensures the order of headings is semantically correct",
    "help": "Heading levels should only increase by one",
    "helpUrl": "https://dequeuniversity.com/rules/axe/4.6/heading-order?application=axeAPI",
    "nodes": [
      {
        "any": [
          {
            "id": "heading-order",
            "data": null,
            "relatedNodes": [],
            "impact": "moderate",
            "message": "Heading order invalid"
          }
        ],
        "all": [],
        "none": [],
        "impact": "moderate",
        "html": "<h4 class=\"Primer_Brand__Heading-module__Heading___IVpmp Primer_Brand__Heading-module__Heading--4___C9jDG Primer_Brand__Accordion-module__Accordion__summary-heading_____snF\">What is GitHub Galaxy?</h4>",
        "target": [
          "details:nth-child(1) > summary > .Primer_Brand__Accordion-module__Accordion__summary-heading_____snF.Primer_Brand__Heading-module__Heading--4___C9jDG"
        ],
        "failureSummary": "Fix any of the following:\n  Heading order invalid"
      }
    ]
  }
```

**Solution:**

This should be customisable using the `as` prop as we do in other components, with `h4` becoming a default value to avoid risk of regressions.
